windows-server-container-tools/CleanupContainerHostNetworking/WindowsContainerNetworking-LoggingAndCleanupAide.ps1

```diff
@@ -0,0 +1,421 @@
+     foreach ($switch in $switches) {
+         if ( $switch.ElementName -eq $switchName) {
+             $ExternalSwitch = $switch
+             break
+         }
+     }
+
+     if (Test-Path "C:\Windows\System32\vfpctrl.exe")
+     {
+         $vfpCtrlExe = "vfpctrl.exe"
+         $ports = $ExternalSwitch.GetRelated("Msvm_EthernetSwitchPort", "Msvm_SystemDevice", $null, $null, $null, $null, $false, $null)
+         foreach ($port in $ports) {
+             $portGuid = $port.Name
+             echo "Policy for port : " $portGuid
+             & $vfpCtrlExe /list-space  /port $portGuid
+             & $vfpCtrlExe /list-mapping  /port $portGuid
+             & $vfpCtrlExe /list-rule  /port $portGuid
+             & $vfpCtrlExe /port $portGuid /get-port-state
+         }
+     }
+ }
+
+# -----------------------------------------------------------------------
+# This function collects logs to capture machine state
+# Parameters: -LogsPath <Path where logs should be saved>
+# Returns: _
+#
+# Information/logs collected by this function:
+#   - HNS Event log: c:\Windows\System32\winevt\Logs\Microsoft-Windows-Host-Network-Service-Admin.evtx
+#   - System event log: c:\Windows\System32\winevt\Logs\System.evtx
+#   - HNS.data file: c:\ProgramData\Microsoft\Windows\HNS\HNS.data
+#   - `ipconfig /allcompartments /all`
+#   - Info for each switch: `GetAllPolicies -switchName $switch.Name`
+#   - VFP info: Get-Service vfpext
+#   - Docker version: `docker -v`
+#   - Docker info: `docker info`
+#   - `Get-VMSwitch`
+#   - `Get-NetNat`
+#   - `Get-NetNatStaticMapping`
+#   - `Get-Service winnat`
+#   - `Get-Service mpssvc`
+#   - `Get-NetAdapter -IncludeHidden`
+#   - `Get-NetAdapterBinding -IncludeHidden`
+#   - `Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\hns`
+#   - `Get-ChildItem HKLM:\SYSTEM\CurrentControlSet\Services\vmsmp`
+# -----------------------------------------------------------------------
+function CopyAllLogs
+{
+    Param(
+      [string] $LogsPath
+    )
+
+    $LogsPath += "_$($script:namingSuffix)"
+
+    if (!(Test-Path $LogsPath))
+    {
+        mkdir $LogsPath
+    }
+
+    Copy-Item c:\Windows\System32\winevt\Logs\Microsoft-Windows-Host-Network-Service-Admin.evtx $LogsPath
+    Copy-Item c:\Windows\System32\winevt\Logs\System.evtx $LogsPath
+
+    $hns = Get-Service hns
+
+    try
+    {
+        if ($hns.Status -eq "Running")
+        {
+            net stop hns
+        }
+
+        if (Test-Path "C:\ProgramData\Microsoft\Windows\HNS\HNS.data")
+        {
+            Copy-Item c:\ProgramData\Microsoft\Windows\HNS\HNS.data $LogsPath
+        }
+    }
+    finally
+    {
+        if ($hns.Status -eq "Running")
+        {
+            net start hns
+        }
+    }
+
+    ipconfig /allcompartments /all > $LogsPath"\ipconfig.txt"
+
+    if (Test-Path "C:\Windows\System32\vfpctrl.exe")
+    {
+        $allSwitches = Get-VMSwitch
```

Comment:
Get-VmSwitch will only work if Hyper-V is installed but Windows Server 2016 container hosts do not need to have Hyper-V installed. If containers role is installed, but Hyper-V is not, this will return a cryptic WMI error.

It might be out of scope for this PR, but this should be cased around Hyper-V role being installed.